app/lib/actions/pulp3/orchestration/repository/upload_content.rb

```diff
@@ -7,17 +8,25 @@ class UploadContent < Pulp3::Abstract
           def plan(repository, smart_proxy, file, unit_type_id)
             sequence do
               content_backend_service = SmartProxy.pulp_primary.content_service(unit_type_id)
-              duplicate_sha_path_content_list = content_backend_service.content_api.list(
+              duplicates_allowed = ::Katello::RepositoryTypeManager.find_content_type(unit_type_id).try(:duplicates_allowed)
+              if repository.generic? && duplicates_allowed
+                filename_key = ::Katello::RepositoryTypeManager.find_content_type(unit_type_id).filename_key
+                duplicate_sha_path_content_list = content_backend_service.content_api(repository.repository_type, unit_type_id).list(
+                  "sha256": file[:sha256],
+                  filename_key => file[:filename])
+              elsif repository.generic?
+                duplicate_sha_path_content_list = content_backend_service.content_api(repository.repository_type, unit_type_id).list(
+                  "sha256": file[:sha256])
```

Comment:
I don't think file[:sha256] is actually present here, notice that it calculates it in the block below